src/neo/SmartContract/Native/ManagementContract.cs

```diff
@@ -0,0 +1,141 @@
+#pragma warning disable IDE0051
+
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using System;
+using System.Numerics;
+
+namespace Neo.SmartContract.Native
+{
+    public sealed class ManagementContract : NativeContract
+    {
+        public override string Name => "Neo Contract Management";
+        public override int Id => 0;
+        public override uint ActiveBlockIndex => 0;
+
+        private const byte Prefix_NextAvailableId = 15;
+        private const byte Prefix_Contract = 8;
+
+        private int GetNextAvailableId(StoreView snapshot)
+        {
+            StorageItem item = snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_NextAvailableId), () => new StorageItem(1));
+            int value = (int)(BigInteger)item;
+            item.Add(1);
+            return value;
+        }
+
+        internal override void OnPersist(ApplicationEngine engine)
+        {
+            foreach (NativeContract contract in Contracts)
+            {
+                if (contract.ActiveBlockIndex != engine.Snapshot.PersistingBlock.Index)
+                    continue;
+                engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_Contract).Add(contract.Hash), new StorageItem(new ContractState
+                {
+                    Id = contract.Id,
+                    Script = contract.Script,
+                    Hash = contract.Hash,
+                    Manifest = contract.Manifest
+                }));
+                contract.Initialize(engine);
+            }
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        internal ContractState GetContract(StoreView snapshot, UInt160 hash)
```

Comment:
```suggestion
        public ContractState GetContract(StoreView snapshot, UInt160 hash)
```
It's called in neo-node.

And maybe we should rename `GetContract` in NativeContract.
https://github.com/neo-project/neo/blob/5b48ee8ad83c98ca4c98c18905eeec934bee3279/src/neo/SmartContract/Native/NativeContract.cs#L94

Reply:
No need to rename. They have different parameter list.